kibicara/platforms/email/webapi.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (C) 2020 by Maike <maike@systemli.org>
+#
+# SPDX-License-Identifier: 0BSD
+
+from fastapi import APIRouter, Depends, HTTPException, status
+from kibicara.platforms.email.bot import spawner
+from kibicara.platforms.email.model import Email, EmailSubscribers
+from kibicara.platformapi import Message
+from kibicara.config import config
+from kibicara.email import send_email
+from kibicara.model import Hood
+from kibicara.webapi.hoods import get_hood
+from ormantic.exceptions import NoMatch
+from pydantic import BaseModel
+from sqlite3 import IntegrityError
+from nacl.encoding import URLSafeBase64Encoder
+from nacl.secret import SecretBox
+from os import urandom
+
+
+class BodyMessage(BaseModel):
+    """ This model shows which values are supplied by the MDA listener script. """
+    text: str
+    to: str
+    author: str
+    secret: str
+
+
+class Subscriber(BaseModel):
+    """ This model holds the email address of a fresh subscriber. """
+    email: str
+
+
+async def get_email_row(to: str):
+    """ Search for Email row if you only have an email address of a bot.
+
+    :param to: email address of a Kibicara hood, e.g. hood@kibicara.org
+    :return: row of Email table, belonging to that email address.
+    """
+    hood_name = to.split('@')[0]
+    hood = await Hood.objects.get(name=hood_name)
+    try:
+        return await Email.objects.get(hood=hood.id)
+    except NoMatch:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND)
+
```

Comment:
This should maybe be moved to kibicara_mda.py; I don't really want to many database calls in that small helper script, but OTOH it's already importing that Table anyway.

Reply:
Another problem is that this doesn't really sanitize the To: email header. I'm not sure if it always has the format `hood@example.com`, or if it's not something like `hood <hood@example.com>` sometimes.